pkg/controller/common/common.go

```diff
@@ -123,18 +123,46 @@ func (controllerContext ControllerContext) Stop() {
 	controllerContext.Broadcaster.Shutdown()
 }
 
-type ChildMap map[string]map[string]*unstructured.Unstructured
+type GroupVersionKind struct {
+	schema.GroupVersionKind
+}
+
+func (gvk GroupVersionKind) MarshalText() ([]byte, error) {
+	var marshalledText string
+	if gvk.Group == "" {
+		marshalledText = fmt.Sprintf("%s.%s", gvk.Kind, gvk.Version)
+	} else {
+		marshalledText = fmt.Sprintf("%s.%s/%s", gvk.Kind, gvk.Group, gvk.Version)
+	}
+	return []byte(marshalledText), nil
+}
+
+func (gvk *GroupVersionKind) UnmarshalText(text []byte) error {
+	kindGroupVersionString := string(text)
+	parts := strings.SplitN(kindGroupVersionString, ".", 2)
+	groupVersion, err := schema.ParseGroupVersion(parts[1])
+	if err != nil {
+		return err
+	}
+	*gvk = GroupVersionKind{
+		groupVersion.WithKind(parts[0]),
+	}
+	return nil
+}
 
-func (m ChildMap) InitGroup(apiVersion, kind string) {
-	m[childMapKey(apiVersion, kind)] = make(map[string]*unstructured.Unstructured)
+type ChildMap map[GroupVersionKind]map[string]*unstructured.Unstructured
```

Comment:
Lets explain this structure via code comments.
If ChildMap is used for children objects as well as for related objects then the term Child should be avoided.

Reply:
Added comment.

Agree with name, but nothing really comes to my mind, only sth like `ObjectMap`, any suggestion ?

Reply:
`RelativeObjectMap` since values are stored relative to parent's namespace.

_As a matter of fact when I designed the GenericController in metac I avoided this concept totally & resorted to independent watch & attachments mapping._

Reply:
good proposal, adjusted code.

> As a matter of fact when I designed the GenericController in metac I avoided this concept totally & resorted to independent watch & attachments mapping.

I do not want to introduce any breaking changes here ;)